packages/synapse-react-client/src/components/CitationPopover/CitationPopover.tsx

```diff
@@ -0,0 +1,335 @@
+      display: 'flex',
+      alignItems: 'center',
+    },
+    '.MuiInputBase-input': {
+      padding: '7px 10px',
+    },
+  }
+
+  const quotationSvg = (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      width="12"
+      height="13"
+      viewBox="0 0 12 13"
+      fill="none"
+    >
+      <path
+        d="M1.2002 6.85162H3.497C3.497 7.82963 3.09981 8.71402 2.4614 9.35362C2.45901 9.356 2.45421 9.3608 2.45179 9.36322V10.3364C3.18859 10.2116 3.84859 9.85881 4.35379 9.35361C4.99457 8.71283 5.38939 7.82961 5.38939 6.85161L5.38943 2.66241H1.20022L1.2002 6.85162Z"
+        fill="#395979"
+      />
+      <path
+        d="M6.61035 2.66242V6.85162H8.90715C8.90715 7.82963 8.50996 8.71402 7.87155 9.35362C7.86917 9.356 7.86436 9.3608 7.86194 9.36322V10.3364C8.59874 10.2116 9.25874 9.85881 9.76394 9.35361C10.4048 8.714 10.7995 7.83081 10.7995 6.85161V2.66241L6.61035 2.66242Z"
+        fill="#395979"
+      />
+    </svg>
+  )
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setCitationFormat(event.target.value)
+  }
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+
+  const handleDownload = () => {
+    if (!citation) {
+      return
+    }
+    const extensionMap: Record<string, string> = {
+      apa: 'ris',
+      bibtex: 'bib',
+      ieee: 'ris',
+      nature: 'ris',
+      science: 'ris',
+    }
+    const citationTitle = title ? `${title.replace(/\s+/g, '_')}` : 'citation'
+    const fileExtension = extensionMap[citationFormat] || 'txt'
+
+    const blob = new Blob([citation], { type: 'text/plain' })
+    const url = URL.createObjectURL(blob)
+    const link = document.createElement('a')
+    link.href = url
+    link.setAttribute('download', `${citationTitle}.${fileExtension}`)
+    document.body.appendChild(link)
+    link.click()
+    document.body.removeChild(link)
+  }
+
+  const handleCopy = () => {
+    if (!citation) {
+      return
+    }
+    navigator.clipboard.writeText(citation)
+  }
+
+  const open = Boolean(anchorEl)
+  const id = open ? 'cite-as-popover' : undefined
+
+  useEffect(() => {
+    const getCitation = async () => {
+      setLoading(true)
+      const formattedDoi = doi ? doi.replace('https://doi.org/', '') : ''
+
+      try {
+        const response = await fetch(
+          `https://citation.doi.org/format?doi=${formattedDoi}&style=${citationFormat}&lang=en-US`,
+          {
+            headers: {
+              Accept: `text/x-${citationFormat}`,
+            },
+          },
+        )
+        if (!response.ok) {
+          throw new Error(
+            `Failed to fetch citation: ${response.status}: ${response.statusText}`,
+          )
+        }
+        const citationText = await response.text()
+        setCitation(citationText)
+      } catch (err) {
+        console.log(err)
+      } finally {
+        setLoading(false)
+      }
+    }
```

Comment:
Can you encapsulate the logic to retrieve a query into a custom hook? Both for testability and reusability with different UI in the future.

Also, please use `useQuery` from `@tanstack/react-query` to fetch data asynchronously, since that will manage providing loading & error states that you can handle in your code.